### ISSUE TITLE
Detected on EAC

### ISSUE THREAD

**Issue by FiYHer** [rezadfx/HWID-shifter#2]
This method was detected a long time ago

**Comment by mapleint** [comment]
You must keep the device controller ptr within the bounds of the .text section of CLASSPNP.SYS. they do a simple integrity check for the function pointer's this way.
You will get banned by any competent rootkit AC for spoofing this way, and it has been this way for roughly 2-3 years now, whenever btbd and namazso published their spoofers.

One way to bypass this is to do some paging magic as vmcall did.
there are certainly more currently undetected ways of spoofing, for instance your EASY-HWID-SPOOFER can be detected not just by its hooks, but by the fact it uses IOCTL (unless you're bypassing DSE).
one last way is just to make the devicecontroller point to some other function.
`	driverObject->MajorFunction[IRP_MJ_DEVICE_CONTROL] = driverObject->MajorFunction[IRP_MJ_QUERY_INFORMATION];`
however I'm pretty sure EAC and BE started detecting this meme about 4-5 months ago.

**Comment by FiYHer** [comment]
> You must keep the device controller ptr within the bounds of the .text section of CLASSPNP.SYS. they do a simple integrity check for the function pointer's this way.
> You will get banned by any competent rootkit AC for spoofing this way, and it has been this way for roughly 2-3 years now, whenever btbd and namazso published their spoofers.
> 
> One way to bypass this is to do some paging magic as vmcall did.
> there are certainly more currently undetected ways of spoofing, for instance your EASY-HWID-SPOOFER can be detected not just by its hooks, but by the fact it uses IOCTL (unless you're bypassing DSE).
> one last way is just to make the devicecontroller point to some other function.
> ` driverObject->MajorFunction[IRP_MJ_DEVICE_CONTROL] = driverObject->MajorFunction[IRP_MJ_QUERY_INFORMATION];`
> however I'm pretty sure EAC and BE started detecting this meme about 4-5 months ago.

yes  im trying two now
1 wpp (there is a small problem, the mapping driver is invalid)
2 physical memory scan serial number and replace

**Comment by mapleint** [comment]
Physically scanning memory for your serial number and replacing it is really a great strategy, I can't think of any detection vectors short of writing a way to grab the serial directly from the HDD and looking for anomalies (EAC and BE are too lazy to do that or it may not even be possible).
As for wpp you can possibly load a 'signed' driver by setting g_CiOptions to 0, then loading your driver conventionally, then setting g_CiOptions back to 1 before patchguard will get you. this race condition will only result in a bugcheck extremely rarely and has the added benefit of giving you the ability to use SEH and IOCTLs.

**Comment by FiYHer** [comment]
> Physically scanning memory for your serial number and replacing it is really a great strategy, I can't think of any detection vectors short of writing a way to grab the serial directly from the HDD and looking for anomalies (EAC and BE are too lazy to do that or it may not even be possible).
> As for wpp you can possibly load a 'signed' driver by setting g_CiOptions to 0, then loading your driver conventionally, then setting g_CiOptions back to 1 before patchguard will get you. this race condition will only result in a bugcheck extremely rarely and has the added benefit of giving you the ability to use SEH and IOCTLs.

yes bro there are still some problems to be solved in the physical memory modification serial number
1. its very slow to modify the hard disk serial number on a real device
2. may cause the system to blue screen and fail to boot
3. modifying the mac will cause the network to be disconnected

wpp can't run depending on the way it is mapped i tried to use the'iqvw64e.sys' i wrote to map wpp. it doesn't seem to work, but it would be dangerous to use a leaked signature certificate

**Comment by mapleint** [comment]
wpp can't run depending on the way it is mapped i tried to use the'iqvw64e.sys' i wrote to map wpp. it doesn't seem to work, but it would be dangerous to use a leaked signature certificate
I think that's just a natural consequence of manual mapped executables, you can't have SEH or any fun with debugging magic.

I would try making a non-signed driver yourself that simply disables DSE and mapping that with some mapper of your choice,
then you can load your driver as if it was signed (when in reality it isn't), then reenable DSE here, then clear the traces of iqvw64e.sys, change its timestamp to a newer less than banned version, and maybe change the size in mmunloaddrivers.
With all of that, you should be able to use wpp, since you are a properly loaded, signed driver.

https://github.com/SamLarenN/CPUZ-DSEFix <- example of changing DSE to load a driver

**Comment by FiYHer** [comment]
> > wpp can't run depending on the way it is mapped i tried to use the'iqvw64e.sys' i wrote to map wpp. it doesn't seem to work, but it would be dangerous to use a leaked signature certificate
> > I think that's just a natural consequence of manual mapped executables, you can't have SEH or any fun with debugging magic.
> 
> I would try making a non-signed driver yourself that simply disables DSE and mapping that with some mapper of your choice,
> then you can load your driver as if it was signed (when in reality it isn't), then reenable DSE here, then clear the traces of iqvw64e.sys, change its timestamp to a newer less than banned version, and maybe change the size in mmunloaddrivers.
> With all of that, you should be able to use wpp, since you are a properly loaded, signed driver.
> 
> https://github.com/SamLarenN/CPUZ-DSEFix <- example of changing DSE to load a driver

doesn't eac focus on detecting drivers that have been loaded but are not properly signed?
drivers without valid signatures should be easy to detect

**Comment by mapleint** [comment]
`doesn't eac focus on detecting drivers that have been loaded but are not properly signed?
drivers without valid signatures should be easy to detect`
I've seen people use this strategy for battleye unsure of EAC since I haven't gotten to reversing that
